app/src/main/java/com/gusta/wakemehome/DetailActivity.java

```diff
@@ -1,246 +1,356 @@
-        }
-
-        return super.onOptionsItemSelected(item);
-    }
-}
+package com.gusta.wakemehome;
+
+import android.content.Context;
+import android.content.ContextWrapper;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.gusta.wakemehome.database.AlarmEntry;
+import com.gusta.wakemehome.databinding.ActivityDetailBinding;
+import com.gusta.wakemehome.maps.MapAddress;
+import com.gusta.wakemehome.maps.MapsActivity;
+import com.gusta.wakemehome.viewmodel.AppExecutors;
+import com.gusta.wakemehome.viewmodel.DetailViewModel;
+
+import java.io.File;
+import java.util.Objects;
+
+public class DetailActivity extends AppCompatActivity {
+
+    //===========//
+    // CONSTANTS //
+    //===========//
+
+    // Constant for logging
+    private static final String TAG = DetailActivity.class.getSimpleName();
+    //default value for alarm id
+    public static final int DEFAULT_ALARM_ID = -1;
+    //Extra alarm id from main activity
+    public static final String EXTRA_ALARM_ID = "extraAlarmId";
+    //extra alarm from map provider
+    public static final String EXTRA_ALARM_COORDINATES = "alarmCoordinates";
+    //ui element which suppose to save on rotate
+    public static final String INSTANCE_ALARM_ID = "instanceAlarmId";
+    public static final String INSTANCE_ALARM_MESSAGE = "instanceAlarmMessage";
+    public static final String INSTANCE_ALARM_ALERT = "instanceAlarmAlert";
+    public static final String INSTANCE_ALARM_VIBRATE = "instanceAlarmVibrate";
+    public static final String INSTANCE_ALARM_ADDRESS_DATA = "instanceAlarmAddressData";
+
+    //temp png will be for unsaved snapshots on save it'll change to  map id.png
+    public static final String TEMP_IMAGE_FILE = "temp.png";
+    // map intent request code
+    private static final int MAP_REQUEST_CODE = 1;
+
+    //=========//
+    // MEMBERS //
+    //=========//
+
+    private AlarmEntry mAlarmEntry;                 // The current alarm entry
+    private DetailViewModel mViewModel;             // The current alarm view model
+    private ActivityDetailBinding mDetailBinding;   // The data binding object
+
+    //=========//
+    // METHODS //
+    //=========//
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mAlarmEntry = new AlarmEntry();
+        mAlarmEntry.setId(DEFAULT_ALARM_ID);

```

Comment:
As I see it, both lines are terrible practice: first saves data on activity, second sets an invalid ID to an entity.